File: app/src/main/java/com/an/github/di/module/DbModule.java
package com.an.github.di.module;


import android.app.Application;
import android.arch.persistence.room.Room;
import android.support.annotation.NonNull;

import com.an.github.data.local.AppDatabase;
import com.an.github.data.local.dao.GithubDao;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DbModule {

    @Provides
    @Singleton
    AppDatabase provideDatabase(@NonNull Application application) {
        return Room.databaseBuilder(application,
                AppDatabase.class, "Github.db")
                .allowMainThreadQueries().build();
    }

    @Provides
    @Singleton
    GithubDao provideGithubDao(@NonNull AppDatabase appDatabase) {
        return appDatabase.githubDao();
    }
}

File: app/src/main/java/com/an/github/di/module/DbModule.kt
package com.an.github.di.module


import android.app.Application
import android.arch.persistence.room.Room

import com.an.github.data.local.AppDatabase
import com.an.github.data.local.dao.GithubDao

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class DbModule {

    @Provides
    @Singleton
    internal fun provideDatabase(application: Application): AppDatabase {
        return Room.databaseBuilder(application,
                AppDatabase::class.java, "Github.db")
                .allowMainThreadQueries().build()
    }

    @Provides
    @Singleton
    internal fun provideGithubDao(appDatabase: AppDatabase): GithubDao {
        return appDatabase.githubDao()
    }
}
